### ISSUE TITLE
"helm upgrade" fails due to init-migrations job conflict

### ISSUE THREAD

**Issue by jkieltyka** [Kong/charts#43]
Helm Version: `version.BuildInfo{Version:"v3.0.3", GitCommit:"ac925eb7279f4a6955df663a0128044a8a6b7593", GitTreeState:"clean", GoVersion:"go1.13.7"}`

Kong Version: Tested against 1.4 and 1.3 Enterprise with PostgresDB enabled

During a helm upgrade --install, with runMigrations set to true, the migration fails due to the init-migrations job. Helm 3 appears to be failing as the job specs are marked as immutable. This failure will still occur with no changes to the values.yaml between upgrades. 
Example command:
`helm upgrade --install kong/kong kong-test  -f ./values.yaml --debug`

Example error:
```bash 
Error: UPGRADE FAILED: cannot patch "kong-test-kong-init-migrations" with kind Job: Job.batch "kong-test-kong-init-migrations" is invalid: spec.template: Invalid value: core.PodTemplateSpec{ObjectMeta:v1.ObjectMeta... }: field is immutable
```

I was able to bypass this error by locally adding `runInitMigrations: false` to the values.yaml file and adding it to the and conditions used to create the job in the template.

**Comment by moconno** [comment]
I'm also seeing the same error when trying to upgrade:
`client.go:183: [debug] error updating the resource "w2g-kong-kong-init-migrations":
	 cannot patch "w2g-kong-kong-init-migrations" with kind Job: Job.batch "w2g-kong-kong-init-migrations" is invalid: spec.template:`

Is there a workaround until this is fixed?  I would like to avoid disabling migrations.

**Comment by hbagdi** [comment]
https://github.com/Kong/charts/pull/36/commits/ae54031aa01f04951054e0723357bf532a7e0eeb fixes the problem but we are blocked by a CI issue. Please see https://github.com/Kong/charts/pull/36.

**Comment by rainest** [comment]
Version 1.3.0 of the Kong chart is [now released](https://github.com/Kong/charts/releases/tag/kong-1.3.0), and includes documentation about this issue in FAQs.md.

A proper fix is contingent on upstream changes to Helm itself, so leaving this open, though changing the title to reflect the actual issue more clearly.

**Comment by rainest** [comment]
Proposed possible solution upstream: https://github.com/helm/helm/issues/7740

**Comment by hbagdi** [comment]
@rainest Can we close this?

**Comment by rainest** [comment]
Yeah, thought PR had closed it automatically. This is fixed in 1.5.0.
